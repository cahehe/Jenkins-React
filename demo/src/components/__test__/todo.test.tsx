import {render, screen, cleanup} from "@testing-library/react"
import renderer from 'react-test-renderer'
import Todo from "../todo"

afterEach(() => {
    cleanup()
})

test('should render non-completed todo component', () => {
    const todo = {id : 1, title : "wash dishes", completed: false}
    const {getByText} = render(<Todo allTodos= {todo} />);    

    // const todoElement = screen.getByTestId('todo-1');

    const todoElement = getByText("wash dishes")
    console.log(todoElement)

    expect(todoElement).toBeInTheDocument()
    expect(todoElement).toHaveTextContent("wash dishes")
    expect(todoElement).not.toContainHTML('<s>');
})


test('should render non-completed todo component', () => {
    const todo = {id : 2, title : "wash dishes", completed: true}
    render(<Todo allTodos= {todo} />);
    const todoElement = screen.getByTestId('todo-2');
    expect(todoElement).toBeInTheDocument();
    expect(todoElement).toHaveTextContent("wash dishes");
    // below does not works since <s> is not a regular html element
    //expect(todoElement).toContainHTML('<s>');
})

test('matchse snapshot', () => {
    const todo = {id : 1, title : "wash dishes", completed: false}
    const tree = renderer.create(<Todo allTodos= {todo} />).toJSON();
    expect(tree).toMatchSnapshot();
})