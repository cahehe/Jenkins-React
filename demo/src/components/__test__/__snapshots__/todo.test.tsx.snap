// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matchse snapshot 1`] = `
<div
  className="todo"
  data-testid="todo-1"
>
  <h1>
    wash dishes
  </h1>
</div>
`;
